log-analyzer/features/org.wso2.carbon.la.apim.feature/src/main/resources/gadgets/LogAPIMArtifactDeleted/js/VizGrammar.js

```diff
@@ -0,0 +1,2300 @@
+        scales.push(colorScale);
+    }
+
+    var axes =  getXYAxes(config, "x", "x", "y", "y");
+
+    marks.push(getLineMark(config, this.metadata));
+    config.markSize = 20;
+    marks.push(getSymbolMark(config, this.metadata));
+
+    if (config.range) {
+        signals = getRangeSignals(config, signals);
+        marks = getRangeMark(config, marks);
+    }
+
+    if (config.color != -1) {
+
+        var legendTitle = "Legend";
+
+        if (config.title != "table") {
+            legendTitle = config.title;
+        }
+
+        var legends = [
+            {
+                "fill": "color",
+                "title": "Legend",
+                "offset": 0,
+                "properties": {
+                    "symbols": {
+                        "stroke": {"value": "transparent"}
+                    },
+                    "title": {
+                        "fill": {"value": config.legendTitleColor},
+                        "fontSize": {"value": config.legendTitleFontSize}
+                    },
+                    "labels": {
+                        "fill": {"value": config.legendTextColor},
+                        "fontSize": {"value": config.ledgendTextFontSize}
+                    }
+                }
+            }
+        ];
+
+        this.spec.legends = legends;
+    }
+
+    this.spec.width = config.width;
+    this.spec.height = config.height;
+    this.spec.axes = axes;
+    this.spec.data = dataTable;
+    this.spec.scales = scales;
+    this.spec.padding = config.padding;
+    this.spec.marks = marks;
+    this.spec.signals = signals;
+
+};
+
+line.prototype.draw = function(div, callbacks) {
+    if(this.config.maxLength != -1){
+        var dataset = this.spec.data[0].values;
+        var maxValue = this.config.maxLength;
+        if(dataset.length >= this.config.maxLength){
+            var allowedDataSet = [];
+            var startingPoint = dataset.length - maxValue;
+            for(var i = startingPoint; i < dataset.length;i++){
+                allowedDataSet.push(dataset[i]);
+            }
+            this.spec.data[0].values = allowedDataSet;
+        }
+    }
+
+    drawChart(div, this, callbacks);
+
+};
+
+line.prototype.insert = function(data) {
+    //Removing events when max value is enabled
+    if (this.config.maxLength != -1 && this.config.maxLength <  (this.view.data(this.config.title).values().length + data.length)) {
+        var removeFunction = (function(d) {
+            return d[this.metadata.names[this.config.x]] == oldData;
+        }).bind(this);
+
+        for (i = 0; i < data.length; i++) {
+            var oldData = this.view.data(this.config.title).values()[i][this.metadata.names[this.config.x]];
+            this.view.data(this.config.title).remove(removeFunction);
+        }
+    }
+
+    this.view.data(this.config.title).insert(data);
+    this.view.update();
+};
+
+line.prototype.getSpec = function() {
+    return this.spec;
+};
+
+function getLineMark(config, metadata){
```

Comment:
Hi Chandana,
We have replace the Vizgrammer.js with Vizgrammer.min.js and relevant commits are in

https://github.com/wso2/shared-analytics/pull/126/files#diff-6ac780d61b2b6d470c8ff04bdeed3bcdR52

Thanks,
SajithD